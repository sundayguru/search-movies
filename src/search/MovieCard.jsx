import React from 'react'


export default class MovieCard extends React.Component {
    render() {
        const {movie} = this.props;
        return (
            <div className="card" style={{backgroundImage: `url(https://image.tmdb.org/t/p/w500${movie.posterPath})`}}>
                <span className="card-label">{movie.adult ? '18+' : 'G'}</span>        
                <div className="card-info">
                    <h2 className="card-title">{movie.title}</h2>
                    <p>Released: {movie.releaseDate}</p>
                </div>
            </div>
        )
    }
}
