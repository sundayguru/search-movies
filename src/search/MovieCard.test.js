import React from 'react';
import MovieCard from './MovieCard';
import renderer from 'react-test-renderer';


test('<MovieCard />', () => {
    const movie = {
        voteAverage: 7.7,
        voteCount: 11713,
        id: 283995,
        mediaType: "movie",
        title: "Guardians of the Galaxy Vol. 2",
        popularity: 45.842,
        posterPath: "/y4MBh0EjBlMuOzv9axM4qJlmhzz.jpg",
        originalLanguage: "en",
        adult: false,
        overview: "The Guardians must fight to keep their newfound family together as they unravel the mysteries of Peter Quill's true parentage.",
        releaseDate: "2017-04-19"
    }
    const component = renderer.create(
        <MovieCard movie={movie} expanded={{}} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    const componentExpanded = renderer.create(
        <MovieCard movie={movie} expanded={{[movie.id]: true}}/>,
    );
    let treeExpanded = componentExpanded.toJSON();
    expect(treeExpanded).toMatchSnapshot();
});