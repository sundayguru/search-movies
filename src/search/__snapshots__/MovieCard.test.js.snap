// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieCard /> 1`] = `
<div
  className="card"
  onClick={[Function]}
  style={
    Object {
      "backgroundImage": "url(https://image.tmdb.org/t/p/w500/y4MBh0EjBlMuOzv9axM4qJlmhzz.jpg)",
    }
  }
>
  <span
    className="card-label"
  >
    G
  </span>
  <div
    className="card-info"
  >
    <h2
      className="card-title"
    >
      Guardians of the Galaxy Vol. 2
    </h2>
    <p>
      Released: 
      2017-04-19
    </p>
    <div
      className="more-details"
    >
      <p
        className="movie-desc"
      >
        The Guardians must fight to keep their newfound family together as they unravel the mysteries of Peter Quill's true parentage.
      </p>
      <p>
        Voting Average: 
        7.7
      </p>
      <p>
        Total Votes: 
        11713
      </p>
      <p>
        Popuplarity: 
        45.842
        %
      </p>
      <p>
        Media Type: 
        movie
        %
      </p>
      <p>
        Language: 
        en
        %
      </p>
    </div>
  </div>
</div>
`;

exports[`<MovieCard /> 2`] = `
<div
  className="card card-expanded"
  onClick={[Function]}
  style={
    Object {
      "backgroundImage": "url(https://image.tmdb.org/t/p/w500/y4MBh0EjBlMuOzv9axM4qJlmhzz.jpg)",
    }
  }
>
  <span
    className="card-label"
  >
    G
  </span>
  <div
    className="card-info"
  >
    <h2
      className="card-title"
    >
      Guardians of the Galaxy Vol. 2
    </h2>
    <p>
      Released: 
      2017-04-19
    </p>
    <div
      className="more-details"
    >
      <p
        className="movie-desc"
      >
        The Guardians must fight to keep their newfound family together as they unravel the mysteries of Peter Quill's true parentage.
      </p>
      <p>
        Voting Average: 
        7.7
      </p>
      <p>
        Total Votes: 
        11713
      </p>
      <p>
        Popuplarity: 
        45.842
        %
      </p>
      <p>
        Media Type: 
        movie
        %
      </p>
      <p>
        Language: 
        en
        %
      </p>
    </div>
  </div>
</div>
`;
